source/AsyncRgbLedSimulationDataGenerator.cpp

```diff
@@ -15,57 +19,97 @@ AsyncRgbLedSimulationDataGenerator::~AsyncRgbLedSimulationDataGenerator()
 
 void AsyncRgbLedSimulationDataGenerator::Initialize( U32 simulation_sample_rate, AsyncRgbLedAnalyzerSettings* settings )
 {
+	// Initialize the random number generator with a literal seed to obtain repeatability
+    // Change this for srand(time(NULL)) for "truly" random sequences
+    // NOTICE rand() an srand() are *not* thread safe
+    srand( 42 );
+
 	mSimulationSampleRateHz = simulation_sample_rate;
 	mSettings = settings;
 
-	mSerialSimulationData.SetChannel( mSettings->mInputChannel );
-	mSerialSimulationData.SetSampleRate( simulation_sample_rate );
-	mSerialSimulationData.SetInitialBitState( BIT_HIGH );
+	mMaximumChannelValue = (1 << mSettings->BitSize()) - 1;
+
+    // TODO pass in the analyzer and call GetMinimumSampleRate?
+    mClockGenerator.Init( 12 * 1000000, mSimulationSampleRateHz );
```

Comment:
this is very minor, but I suggest we pass in 1 as the frequency, for clarity, since it isn't used, and add a comment stating that the clock frequency is never used.

Reply:
No problem, will do.